flows/waits/time.go

```diff
@@ -0,0 +1,25 @@
+package waits
+
+import (
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/events"
+)
+
+const TypeTime string = "time"
+
+type TimeWait struct {
+	Timeout int `json:"timeout" validation:"required,gte=0"`
+}
+
+func (w *TimeWait) Type() string { return TypeMsg }
+
+func (w *TimeWait) Apply(run flows.FlowRun, step flows.Step) {
+	run.ApplyEvent(step, nil, &events.TimeWaitEvent{Timeout: w.Timeout})
+}
+
+// CanResume returns true for a message wait if a message has now been received on this step
+func (w *TimeWait) CanResume(run flows.FlowRun, step flows.Step) bool {
+	return true
```

Comment:
Definitely thing 1) is valid, though this is a TimeWait not a NullWait.

For 2) the engine has to satisfy the contract of the flow and should make sure that isn't violated regardless of what the caller does. So to me it sure seems we need to be checking these things. Agree there can be races with waits but thats the world we live in.